interfaces/cython/cantera/reaction.pyx

```diff
@@ -24,32 +29,121 @@ cdef class _ReactionRate:
         will raise an exception.
         """
         if pressure:
-            self.base.update(temperature, pressure)
-            return self.base.eval(temperature, pressure)
+            self.rate.update(temperature, pressure)
+            return self.rate.eval(temperature, pressure)
         else:
-            self.base.update(temperature)
-            return self.base.eval(temperature)
+            self.rate.update(temperature)
+            return self.rate.eval(temperature)
+
+    @staticmethod
+    cdef wrap(shared_ptr[CxxReactionRateBase] rate):
+        """
+        Wrap a C++ Reaction object with a Python object of the correct derived type.
+        """
+        # ensure all reaction types are registered
+        if not(_reaction_rate_class_registry):
+            def register_subclasses(cls):
+                for c in cls.__subclasses__():
+                    rate_type = getattr(c, "_reaction_rate_type")
+                    _reaction_rate_class_registry[rate_type] = c
+                    register_subclasses(c)
+
+            # update global reaction class registry
+            register_subclasses(ReactionRate)
+
+        # identify class
+        rate_type = pystr(rate.get().type())
+        cls = _reaction_rate_class_registry.get(rate_type, ReactionRate)
+
+        # wrap C++ reaction rate
+        cdef ReactionRate rr
+        rr = cls(init=False)
+        rr._rate = rate
+        rr.rate = rr._rate.get()
+        return rr
+
+    @classmethod
+    def from_dict(cls, data, Kinetics kinetics=None):
+        """
+        Create a `ReactionRate` object from a dictionary corresponding to its YAML
+        representation.
+
+        An example for the creation of a ReactionRate from a dictionary is::
+
+            rate = ReactionRate.from_dict(
+                {"rate-constant": {"A": 38.7, "b": 2.7, "Ea": 26191840.0}},
+                kinetics=gas)
+
+        In the example, *gas* is a Kinetics (or Solution) object.
+
+        :param data:
+            A dictionary corresponding to the YAML representation.
+        :param kinetics:
+            A `Kinetics` object whose associated phase(s) contain the species
+            involved in the reaction.
+        """
+        if cls._reaction_rate_type != "":
+            raise TypeError(
+                "Class method 'from_dict' was invoked from '{}' but should "
+                "be called from base class 'Reaction'".format(cls.__name__))
+        if kinetics is None:
+            raise ValueError("A Kinetics object is required.")
```

Comment:
Correct, but on the other hand, the required `Kinetics` object is a wart that I hope to address at some point.

Reply:
> In other words, I hope to make it optional eventually.

Then, IMO, we should make it optional at that point. This would lead to pretty confusing documentation, where the value would appear to be optional but then actually it's required?

Reply:
Interestingly, some of this turns out to be pre-existing (see `Reaction.fromYaml`. I don't have a problem making all those required.